lib/engine.py

```diff
@@ -348,17 +350,18 @@ def create_table(self):
             print("Couldn't create table (%s). Trying to continue anyway." % e)
 
     def create_table_statement(self):
-        """Returns a SQL statement to create a table."""
-        create_stmt = "CREATE TABLE " + self.table_name() + " ("
-
-        columns = self.table.get_insert_columns(join=False)
+        """Returns a SQL statement to create a table
 
+        use create=True to get insert columns including Auto field for creating table
+        default is false and used when returning only the colums for inserts 
+        """
+        create_stmt = "CREATE TABLE " + self.table_name() + " ("
+        columns = self.table.get_insert_columns(join=False, create=True)
```

Comment:
Is this this only use of `create=True`? If so, instead of putting the new code in `get_insert_columns`, which then requires explaining it here, why not just put the code here? E.g., something like

```
columns = self.table.get_insert_columns(join=False)
if self.columns[0][0] == 'record_id' and columns[0] == 'record_id':
    del columns[0]
```

I'm not sure if we need both checks, but this should, I think, replicate what the code is currently doing.